### PR TITLE
3.6 view request response

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/View/View.php [hunk 1]
```diff
@@ -386,6 +388,66 @@ public function initialize()
     {
     }
 
+    /**
+     * Gets the request instance.
+     *
+     * @return \Cake\Http\ServerRequest
+     * @since 3.6.0
```
Comment (dereuromark) [inline src/View/View.php]: 3.6.3?

**Comment by dereuromark** [comment]
Looks good to me.

**Comment by inoas** [comment]
I always consider(ed) Cells to be mini-controllers, so getters are what I expect, setters not so sure.

**Comment by ADmad** [comment]
> I always consider(ed) Cells to be mini-controllers, so getters are what I expect, setters not so sure.

Can you provide a reasonable use case for getting request instance from cell? Cell are used in controller or view context which already have access to request/response instance.

**Comment by ADmad** [comment]
@markstory I thought it would be best to get as many deprecations as possible out of the way in 3.6. But I am fine re-targetting this to 3.7.

**Comment by markstory** [comment]
@ADmad Is the goal to add deprecation warnings to View for 3.6 as well?

**Comment by ADmad** [comment]
So should I re-target this to 3.7? I don't care either way.

File: src/View/View.php [hunk 2]
```diff
@@ -386,6 +388,66 @@ public function initialize()
...
+    {
+        return $this->response;
+    }
+
+    /**
+     * Sets the response instance.
+     *
+     * @param \Cake\Http\Response $response Response instance.
+     * @return $this
+     * @since 3.6.3
```
[the remark this answers is not in the document]
Comment (inoas) [reply src/View/View.php]: In general I'd love to see the project becoming a bit more strict in this regard!
However, I thought as of 3.6.x as the 3.x LTS but apparently you are prepping 3.7. I wonder why... though. Why not keep 3.6.x as the LTS - that way once a software project is compatible with latest 3.6.x one can enable deprecation warnings and then work through these to upgrade to 4.x, or stay on 3.6 forever until EOL.

**Comment by ADmad** [comment]
This should be good to go.

**Comment by markstory** [comment]
I'll merge this when I get the migration guide updated.

**Comment by ADmad** [comment]
Docs PR is merged :)

**Pull request merged**